### DISCUSSION TITLE
about production deployment connection.

### DISCUSSION THREAD

**Discussion by ice-wizschool** [geckosio/geckos.io#173] [Help / Questions]
I'm using geckos to connect from client to server. 

When I test on localhost with node index.js, the connection works perfectly. 
But when I deploy the code to production server with SSL support, client receiver 404 (not found). 

I was following the document to implement connect, did I miss something?

Btw, I opened all UDP / TCP related port and client directly connect to Geckos code.

**Comment by ouya99** [comment]
@ice-wizschool  did you make it work? same problem here

**Comment by bcjordan** [comment]
Having the same issue finding the right production config. It works when running in http / udp on a server with all ports open, but can't seem to find the right combination of config to work when using https (whether using Fly.io or proxying the working server via Cloudflare). Anyone successfully do this / have an example of how their server's ports / connection are configured?

A sample `fly.toml` config for the client+server phaser-on-nodejs sample repo would be epic!

**Reply by yandeu** [reply]
You only need https for the geckos.io port (9208), WebRTC over UPD is always secure.

If you set a port range (https://github.com/geckosio/geckos.io#custom-port-range), you can simply just open these ports + 9208.

Here is an express + https example that should work:  
https://github.com/geckosio/express-https-test

I don't know/use fly.io

Btw, I recommend [arcade-physics](https://github.com/yandeu/arcade-physics) over  phaser-on-nodejs.

**Reply by bcjordan** [reply]
Thanks for the tip on arcade-physics!

I found where I was missing a 1444 --> 443 port change in the phaser-on-nodejs example, which got it working using a DigitalOcean droplet --> Cloudflare for automatic 80 --> 443 HTTPS.

I've been trying to get it running on Fly.io as well (Docker-based deployment tool with a nice free tier and support for UDP), but due to how they do global routing they have a [constraint where you need to bind the UDP server to `fly-global-services`](https://fly.io/docs/app-guides/udp-and-tcp/#the-fly-global-services-address) instead of `127.0.0.1` (or `localhost` or `*` etc.). I can't seem to find where that might be configurable in `geckos.io` or even the `node-datachannel` wrapper. Any idea if that might be possible to configure that or how to add such a config option down the stack? (Similar to the min/max ports options for the UDP server)

**Reply by bcjordan** [reply]
Ah, believe this is where the config parameter would need to be added upstream to be able to drill through: https://github.com/murat-dogan/node-datachannel/issues/143
